modules/SequencesExt.tla

```diff
@@ -46,6 +46,18 @@ SetToSortSeq(S, op(_,_)) ==
   \* because this variant works efficiently without a dedicated TLC override.
   SortSeq(SetToSeq(S), op)
 
+SetToAllKPermutations(S) ==
+  (**************************************************************************)
+  (* Convert the set S to a set containing all k-permutations for           *)
```

Comment:
I find "containing all k-permutations" a bit vague, perhaps "containing all k-permutations of elements of S"?